File: gfg/2024_7_july/c++/29_row_with_max_1.cpp
// TIME COMPLEXITY
// O(n*m)

// SPACE COMPLEXITY
// O(1)

// CODE

class Solution {
  public:
    int rowWithMax1s(vector<vector<int> > arr) {
        int row = -1;
        int maxcnt = 0;
        
        int n = arr.size();
        int m = arr[0].size();
        for(int i = 0; i < n; i++){
            int cnt = 0;
            for(int j = m-1; j >= 0; j--){
                if(arr[i][j] == 0) break;
                cnt++;
            }
            if(cnt > maxcnt){
                maxcnt = cnt;
                row = i;
            }
        }
        return row;
    }
};

File: gfg/2024_7_july/python/29_row_with_max_1.py
# CODE

class Solution:
    def rowWithMax1s(self, arr):
        m, n = len(arr), len(arr[0])
        ind, res, cnt = 0, 0, n - 1
        while ((ind != m) and (cnt >= 0)):
            if arr[ind][cnt]:
                res = ind
                cnt -= 1
            else:
                ind += 1
        return -1 if cnt == n - 1 else res